agents/craftassist/craftassist_agent.py

```diff
@@ -270,11 +273,29 @@ def perceive(self, force=False):
             # perceive from heuristic perception module
             heuristic_perception_output = self.perception_modules["heuristic"].perceive()
             self.memory.update(heuristic_perception_output)
-        # 4. if semantic segmentation model is initialized, call perceive
-        if "semseg" in self.perception_modules:
-            sem_seg_perception_output = self.perception_modules["semseg"].perceive()
-            self.memory.update(sem_seg_perception_output)
-        self.areas_to_perceive = []
+        # 4. If detection model is initialized and text_span for reference object exists in 
+        # logical form, call perceive().
+        if "detection_model" in self.perception_modules and ref_obj_spans:
+            # FIXME . notation for triple walk
+            chat_memids, _ = self.memory.basic_search(
```

Comment:
I did the searches here to avoid having to return ref_obj_spans from the main droidlet agent's perceive (trying to keep it cleaner).  do we need these if we are returning ref_obj_spans from super.perceive()?  no need to do the searches now I think.